compiler/rustc_middle/src/ty/relate.rs

```diff
@@ -596,6 +596,13 @@ pub fn super_relate_consts<'tcx, R: TypeRelation<'tcx>>(
         );
     }
 
+    // HACK(consts): We still need to eagerly evaluate consts when
+    // relating them because during `normalize_param_env_or_error`,
+    // we may relate an evaluated constant in a obligation against
+    // an unnormalized (i.e. unevaluated) const in the param-env.
```

Comment:
```suggestion
    // HACK(const_generics): We still need to eagerly evaluate consts when
    // relating them because during `normalize_param_env_or_error`,
    // we may relate an evaluated constant in a obligation against
    // an unnormalized (i.e. unevaluated) const in the param-env.
    // FIXME(generic_const_exprs): Once we always lazily unify unevaluated constants
    // these `eval` calls can be removed.
```

Reply:
that's also probably broken with `feature(generic_const_exprs)` because with that constants can refer to higher ranked lifetimes and some type relations don't actually convert bound variables to placeholders/infer when unifying, so this would cause an ICE.

Can you change this to only run if gce is not enabled.